Dockerfile

```diff
@@ -8,8 +8,23 @@ RUN go mod download
 COPY . .
 RUN go build -ldflags="-X 'github.com/kanopy-platform/buildah-plugin/internal/version.version=${VERSION}' -X 'github.com/kanopy-platform/buildah-plugin/internal/version.gitCommit=${GIT_COMMIT}' -X 'github.com/kanopy-platform/buildah-plugin/internal/version.pluginType=${PLUGIN_TYPE}'" -o /go/bin/app ./cmd/
 
+FROM debian:bookworm-slim as ecr-login
+ARG ECR_LOGIN_VERSION="0.7.1"
+ARG ARCH="amd64"
+RUN apt-get update && apt-get install -y wget
+RUN wget -O /docker-credential-ecr-login https://amazon-ecr-credential-helper-releases.s3.us-east-2.amazonaws.com/${ECR_LOGIN_VERSION}/linux-${ARCH}/docker-credential-ecr-login
+RUN chmod +x /docker-credential-ecr-login
+
 FROM quay.io/buildah/stable:v1.31.0
 RUN groupadd -r app && useradd --no-log-init -r -g app app
+# Add ranges needed for buildah commands
+# RUN usermod --add-subuids 10000-65536 app
+# RUN usermod --add-subgids 10000-65536 app
+# Create directory needed to store credentials
+ENV HOME=/buildah
+RUN mkdir -m 777 -p $HOME/.docker
+RUN chown -R app $HOME
```

Comment:
Oh thanks, I didn't bother looking at the upstream `quay.io/buildah/stable` Containerfile till now. Good catch.

Updated to use `USER build` because it seems the reference image already sets up all the necessary permissions subuid, subgid for that user so we don't need to repeat for another user.